aiida/orm/nodes/data/folder.py

```diff
@@ -19,3 +19,22 @@
 
 class FolderData(Data):
     """`Data` sub class to represent a folder on a file system."""
+
+    def __init__(self, **kwargs):
+        """Construct a new `FolderData` to which any files and folders can be added.
+
+        To add files to a new node use the various repository methods:
+
+            folder = FolderData()
+            folder.put_object_from_tree('/absolute/path/to/directory')
+            folder.put_object_from_filepath('/absolute/path/to/file.txt')
+            folder.put_object_from_filelike(filelike_object)
+
+        Alternatively, in order to simply wrap a directory, the `path` keyword can be used in the constructor:
+
+            folder = FolderData(tree='/absolute/path/to/directory')
```

Comment:
Would it make sense to add an explicit documentation of the `tree` argument as well?
```
    :param tree: Absolute path to a folder to wrap.
    :type tree: str
```